Sources/swift-format/CommandLineOptions.swift

```diff
@@ -33,6 +33,11 @@ struct CommandLineOptions {
   /// If specified, the current file is overwritten when formatting
   var inPlace: Bool = false
 
+  /// The directory in which to run the formatter/linter recursively.
+  ///
+  /// If set, we recursively run on all ".swift" files in the provided directory.
+  var recursive: String? = nil
```

Comment:
My thinking for the `-r` flag was that instead of taking a directory name as a value for that argument, it would be a Boolean that "unlocks" the ability to pass directory paths (in addition to file paths) as regular positional arguments. Essentially the same way `grep -r` works; it errors if there's a directory on the command line unless the flag is provided.

So, the usage would be like this:

```
$ swift-format some/directory
error: some/directory is a directory; use -r to recursively lint/format its contents

$ swift-format -r some/directory
# good

$ swift-format -r some/directory SomeFile.swift some/OtherFile.swift
# also good, we recursively lint/format some/directory but we also do the two files

$ swift-format -r some/directory some/directory2
# also good, we recursively lint/format both dirs
```

One could argue that the `-r` option isn't needed in this case because we could just detect directories without the flag, but I think it's nice to have the extra safety net in case you accidentally put a directory in there that you didn't mean to and then "oh crap, just formatted everything in it".

Reply:
Done.